### ISSUE TITLE
Prevent the header overlapping the main content when the banners are visible and skip-to-content is clicked

### ISSUE THREAD

**Issue by obulat** [WordPress/openverse#2603]
## Description

In #2538 we've added simplified skip-to-content links. The problem is that when the users click on the links, the page scrolls under the header. To prevent that, we've added the `scroll-mt-20` class. However, when the banners are visible, the header area becomes taller, and the content is still hidden beneath the header when you click on the "skip-to-content" link.

## Solution
We should use a `--header-height` css variable that is computed in the layouts to account for all of the banners and the header. This CSS variable should be used to set the `scroll-margin` of the `skip-to-content` link.

## Additional information
Originally noted in 2538 by @sarayourfriend :
> LGTM. The changes to prevent the header overlap work when there are no banners. With one banner at least part of the word is still visible, so while not ideal, there's the visual queue to scroll up a little to see the heading. Ideally we'd be able to correct the amount of margin for the number of banners present. A CSS custom property that gets updated by the banner components when they render is the only way I can think to do that, but it would need to be a manual approach rather than using the tailwind utilities for this.

_Originally posted by @sarayourfriend in https://github.com/WordPress/openverse/pull/2538#pullrequestreview-1521222386_
